src/lsm/scan_fuzz.zig

```diff
@@ -0,0 +1,1170 @@
+const ForestType = @import("forest.zig").ForestType;
+const ScanLookupType = @import("scan_lookup.zig").ScanLookupType;
+const TimestampRange = @import("timestamp_range.zig").TimestampRange;
+const Direction = @import("../direction.zig").Direction;
+
+const Grid = GridType(Storage);
+const SuperBlock = vsr.SuperBlockType(Storage);
+
+const batch_max: u32 = @divFloor(
+    constants.message_body_size_max,
+    @sizeOf(Thing),
+);
+
+/// The testing object.
+const Thing = extern struct {
+    id: u128,
+    // All indexes must be `u64` to avoid conflicting matching values,
+    // the most significant bits are a seed used for assertions.
+    index_1: u64,
+    index_2: u64,
+    index_3: u64,
+    index_4: u64,
+    index_5: u64,
+    index_6: u64,
+    index_7: u64,
+    index_8: u64,
+    index_9: u64,
+    index_10: u64,
+    index_11: u64,
+    index_12: u64,
+    checksum: u64,
+    timestamp: u64,
+
+    comptime {
+        assert(stdx.no_padding(Thing));
+        assert(@sizeOf(Thing) == 128);
+        assert(@alignOf(Thing) == 16);
+    }
+
+    /// Initializes a struct with all fields zeroed.
+    fn zeroed() Thing {
+        return .{
+            .id = 0,
+            .index_1 = 0,
+            .index_2 = 0,
+            .index_3 = 0,
+            .index_4 = 0,
+            .index_5 = 0,
+            .index_6 = 0,
+            .index_7 = 0,
+            .index_8 = 0,
+            .index_9 = 0,
+            .index_10 = 0,
+            .index_11 = 0,
+            .index_12 = 0,
+            .checksum = 0,
+            .timestamp = 0,
+        };
+    }
+
+    /// Gets the field's value based on the `Index` enum.
+    fn get_index(thing: *const Thing, index: Index) u64 {
+        switch (index) {
+            inline else => |comptime_index| {
+                return @field(thing, @tagName(comptime_index));
+            },
+        }
+    }
+
+    /// Sets the field's value based on the `Index` enum.
+    fn set_index(thing: *Thing, index: Index, value: u64) void {
+        switch (index) {
+            inline else => |comptime_index| {
+                @field(thing, @tagName(comptime_index)) = value;
+            },
+        }
+    }
+
+    /// Merges all non-zero fields.
+    fn merge(template: *Thing, other: Thing) void {
+        stdx.maybe(stdx.zeroed(std.mem.asBytes(template)));
+        assert(!stdx.zeroed(std.mem.asBytes(&other)));
+        defer assert(!stdx.zeroed(std.mem.asBytes(template)));
+
+        for (std.enums.values(Index)) |index| {
+            const value = other.get_index(index);
+            if (value != 0) {
+                assert(template.get_index(index) == 0);
+                template.set_index(index, value);
+            }
+        }
+    }
+
+    /// Creates a struct from a template, setting all zeroed fields with random values and
+    /// calculating the checksum of the resulting struct.
+    fn from_template(template: Thing, random: std.rand.Random, id: u128, timestamp: u64) Thing {
```

Comment:
usually, when we have  several numeric arguments, we pack them into an options struct:

```
options: struct { id: u128, timestamp: u64}
```

that way its harder to mix them up. In this case, I think it'll also help call-site readability, as it is non-trivial observaion that we use object count as our timestamp.